airflow/www/views.py

```diff
@@ -3083,6 +3083,8 @@ class GraphForm(DateTimeWithNumRunsWithDagRunsForm):
         for ti in dag.get_task_instances(dttm, dttm):
             if ti.task_id not in task_instances:
                 task_instances[ti.task_id] = wwwutils.get_instance_with_map(ti, session)
+                # Need to add operator_name explicitly because it's not a column in task_instances model.
+                task_instances[ti.task_id]["operator_name"] = ti.operator_name
```

Comment:
I think to avoid this we can replace the @property by an sqlalchemy `column_property`. (Would also allow filtering, and querying against it easily)

Reply:
https://docs.sqlalchemy.org/en/20/orm/mapped_sql_expr.html#using-column-property

Reply:
@pierrejeambrun I believe to do this we would do something like this 
```
operator_name = column_property(
        select(
            func.IF(custom_operator_name is not None, custom_operator_name, operator)
        )
    )
```
But the IF statement is not working with sqlite3. Getting below error - 
![Screenshot 2023-06-15 at 10 00 57 AM](https://github.com/apache/airflow/assets/13021213/406d9af2-4c4a-4722-b809-59287a0a403f)

Is there any other way to accomplish this?

Reply:
I think `CASE` is better supported across databases

Reply:
I'll give that a shot.

Reply:
@uranusjr @pierrejeambrun using `column_property` raises below issue 
`AttributeError: Neither 'Label' object nor 'Comparator' object has an attribute 'nullable'` when looking for a solution ran into this thread - https://github.com/marshmallow-code/marshmallow-sqlalchemy/issues/97

As mentioned in the thread it's fixed in `0.28.0` But we cannot upgrade to `0.28.0` because of the below issue - 
`flask-appbuilder 4.3.1 requires marshmallow-sqlalchemy<0.27.0,>=0.22.0,`
flask-appbuilder is pinned to 4.3.1 here - 
https://github.com/apache/airflow/blob/fe7a1aa0962538a99f5e73dbd3890a6f210d22ed/setup.py#L292

Unfortunately even in the latest version of flask-appbuilder 4.3.2 we still have upperbound on `marshmallow-sqlalchemy>=0.22.0, <0.27.0`
https://github.com/dpgaspar/Flask-AppBuilder/blob/9090d096f697cf03dd80ebc656a6b0e79a05adb3/setup.py#LL61

Reply:
The thread mentions adding an explicit field. Maybe this can be worked around by adding an `operator_name` field in the Marshmallow model?

https://github.com/marshmallow-code/marshmallow-sqlalchemy/issues/97#issuecomment-522765892

Reply:
@uranusjr AFAIK, I don't think we are generating `Marshmallow` models directly, it's done `flask_appbuilder/api/convert.py`. Is there any other way that I'm not aware of?